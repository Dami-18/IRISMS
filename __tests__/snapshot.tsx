import { render } from '@testing-library/react'
import WelcomePage from '@/app/page'

it('renders homepage unchanged', () => {
  const { container } = render(<WelcomePage />)
  expect(container).toMatchSnapshot()
})