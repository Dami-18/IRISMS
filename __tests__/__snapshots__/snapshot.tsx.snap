// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePage renders homepage unchanged 1`] = `
<div>
  <div
    class="jsx-723bb9ff0809bb37 min-h-screen bg-gradient-to-br from-yellow-50 to-purple-100"
  >
    <header
      class="jsx-723bb9ff0809bb37 bg-white shadow-md py-4 px-6 flex justify-between items-center"
    >
      <div
        class="jsx-723bb9ff0809bb37 text-3xl font-extrabold text-yellow-600 transform transition duration-500 hover:scale-105"
      >
        IRISMS
      </div>
      <div
        class="jsx-723bb9ff0809bb37 flex space-x-4"
      >
        <button
          class="jsx-723bb9ff0809bb37 px-4 py-2 bg-indigo-600 text-white rounded-lg shadow-md hover:bg-yellow-600 transform transition-all duration-300 hover:-translate-y-1 hover:shadow-lg"
        >
          Login
        </button>
        <button
          class="jsx-723bb9ff0809bb37 px-4 py-2 bg-yellow-600 text-white rounded-lg shadow-md hover:bg-indigo-700 transform transition-all duration-300 hover:-translate-y-1 hover:shadow-lg"
        >
          Register
        </button>
      </div>
    </header>
    <section
      class="jsx-723bb9ff0809bb37 relative bg-indigo-900 text-white py-20 overflow-hidden"
      id="main-header"
    >
      <div
        class="jsx-723bb9ff0809bb37 absolute inset-0 overflow-hidden"
      >
        <div
          class="jsx-723bb9ff0809bb37 absolute inset-0 opacity-50"
        >
          <div
            class="jsx-723bb9ff0809bb37 absolute top-1/4 left-1/4 w-72 h-72 bg-purple-400 rounded-full mix-blend-multiply filter blur-3xl opacity-70 animate-pulse"
          />
          <div
            class="jsx-723bb9ff0809bb37 absolute top-1/3 right-1/4 w-72 h-72 bg-yellow-300 rounded-full mix-blend-multiply filter blur-3xl opacity-70 animate-pulse delay-700"
          />
          <div
            class="jsx-723bb9ff0809bb37 absolute bottom-1/4 right-1/3 w-72 h-72 bg-pink-300 rounded-full mix-blend-multiply filter blur-3xl opacity-70 animate-pulse delay-1000"
          />
        </div>
      </div>
      <div
        class="jsx-723bb9ff0809bb37 container mx-auto px-6 relative z-10"
      >
        <div
          class="jsx-723bb9ff0809bb37 max-w-3xl mx-auto text-center transform transition-all duration-1000 translate-y-0 opacity-100"
        >
          <h1
            class="jsx-723bb9ff0809bb37 text-5xl font-bold mb-6 leading-tight"
          >
            Discover Your Research Journey
          </h1>
          <p
            class="jsx-723bb9ff0809bb37 text-xl mb-8 text-yellow-100"
          >
            Connect with leading professors, find research internships, and explore scholarship opportunities.
          </p>
        </div>
      </div>
    </section>
    <section
      class="jsx-723bb9ff0809bb37 py-16 bg-white"
      id="academic-journey"
    >
      <div
        class="jsx-723bb9ff0809bb37 container mx-auto px-6"
      >
        <h2
          class="jsx-723bb9ff0809bb37 text-3xl font-bold text-center mb-12 text-gray-800"
        >
          How We Help In Your Academic Journey
        </h2>
        <div
          class="jsx-723bb9ff0809bb37 grid md:grid-cols-3 gap-12"
        >
          <div
            class="jsx-723bb9ff0809bb37 bg-blue-50 rounded-xl p-8 shadow-lg transform transition-all duration-500 hover:-translate-y-2 hover:shadow-xl"
          >
            <div
              class="jsx-723bb9ff0809bb37 w-16 h-16 mx-auto bg-blue-500 text-white rounded-lg flex items-center justify-center mb-6"
            >
              <svg
                class="jsx-723bb9ff0809bb37 h-8 w-8"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="jsx-723bb9ff0809bb37"
                  d="M19 21V5a2 2 0 00-2-2H7a2 2 0 00-2 2v16m14 0h2m-2 0h-5m-9 0H3m2 0h5M9 7h1m-1 4h1m4-4h1m-1 4h1m-5 10v-5a1 1 0 011-1h2a1 1 0 011 1v5m-4 0h4"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
            <h3
              class="jsx-723bb9ff0809bb37 text-xl font-semibold text-center mb-3"
            >
              Connect with Professors
            </h3>
            <p
              class="jsx-723bb9ff0809bb37 text-gray-600 text-center"
            >
              Explore profiles of leading professors and researchers across various disciplines and universities worldwide.
            </p>
          </div>
          <div
            class="jsx-723bb9ff0809bb37 bg-purple-50 rounded-xl p-8 shadow-lg transform transition-all duration-500 hover:-translate-y-2 hover:shadow-xl"
          >
            <div
              class="jsx-723bb9ff0809bb37 w-16 h-16 mx-auto bg-purple-500 text-white rounded-lg flex items-center justify-center mb-6"
            >
              <svg
                class="jsx-723bb9ff0809bb37 h-8 w-8"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="jsx-723bb9ff0809bb37"
                  d="M21 13.255A23.931 23.931 0 0112 15c-3.183 0-6.22-.62-9-1.745M16 6V4a2 2 0 00-2-2h-4a2 2 0 00-2 2v2m4 6h.01M5 20h14a2 2 0 002-2V8a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
            <h3
              class="jsx-723bb9ff0809bb37 text-xl font-semibold text-center mb-3"
            >
              Research Internships
            </h3>
            <p
              class="jsx-723bb9ff0809bb37 text-gray-600 text-center"
            >
              Find and apply for research internships that match your academic interests and career goals.
            </p>
          </div>
          <div
            class="jsx-723bb9ff0809bb37 bg-green-50 rounded-xl p-8 shadow-lg transform transition-all duration-500 hover:-translate-y-2 hover:shadow-xl"
          >
            <div
              class="jsx-723bb9ff0809bb37 w-16 h-16 mx-auto bg-green-500 text-white rounded-lg flex items-center justify-center mb-6"
            >
              <svg
                class="jsx-723bb9ff0809bb37 h-8 w-8"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="jsx-723bb9ff0809bb37"
                  d="M12 8c-1.657 0-3 .895-3 2s1.343 2 3 2 3 .895 3 2-1.343 2-3 2m0-8c1.11 0 2.08.402 2.599 1M12 8V7m0 1v8m0 0v1m0-1c-1.11 0-2.08-.402-2.599-1M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
            <h3
              class="jsx-723bb9ff0809bb37 text-xl font-semibold text-center mb-3"
            >
              Scholarship Opportunities
            </h3>
            <p
              class="jsx-723bb9ff0809bb37 text-gray-600 text-center"
            >
              Discover scholarships and grants to fund your education and research endeavors from different organizations.
            </p>
          </div>
        </div>
      </div>
    </section>
    <section
      class="jsx-723bb9ff0809bb37 py-16 bg-gray-50"
    >
      <div
        class="jsx-723bb9ff0809bb37 container mx-auto px-6"
      >
        <h2
          class="jsx-723bb9ff0809bb37 text-3xl font-bold text-center mb-4 text-gray-800"
        >
          Featured Professors
        </h2>
        <p
          class="jsx-723bb9ff0809bb37 text-center text-gray-600 mb-12 max-w-2xl mx-auto"
        >
          Connect with leading researchers and professors who are at the forefront of innovation and discovery.
        </p>
        <div
          class="jsx-723bb9ff0809bb37 grid grid-cols-2 md:grid-cols-4 gap-6"
        >
          <div
            class="jsx-723bb9ff0809bb37 bg-white rounded-lg shadow-md p-6 flex flex-col items-center text-center transform transition-all duration-500 hover:-translate-y-2 hover:shadow-xl"
          >
            <div
              class="jsx-723bb9ff0809bb37 w-24 h-24 rounded-full bg-gray-200 mb-4 overflow-hidden"
            >
              <img
                alt="Prof. Debasis Samanta"
                class="jsx-723bb9ff0809bb37 w-full h-full object-cover"
                src="https://cse.iitkgp.ac.in/facimg/DSMsmall.jpg?1743340966"
              />
            </div>
            <h3
              class="jsx-723bb9ff0809bb37 font-medium text-lg mb-1"
            >
              Prof. Debasis Samanta
            </h3>
            <p
              class="jsx-723bb9ff0809bb37 text-sm text-gray-500 mb-2"
            >
              Biometric Security, Brain Computing, Human Computer Interaction
            </p>
          </div>
          <div
            class="jsx-723bb9ff0809bb37 bg-white rounded-lg shadow-md p-6 flex flex-col items-center text-center transform transition-all duration-500 hover:-translate-y-2 hover:shadow-xl"
          >
            <div
              class="jsx-723bb9ff0809bb37 w-24 h-24 rounded-full bg-gray-200 mb-4 overflow-hidden"
            >
              <img
                alt="Prof. Jibesh Patra"
                class="jsx-723bb9ff0809bb37 w-full h-full object-cover"
                src="https://cse.iitkgp.ac.in/facimg/JBPsmall.jpg?1743340966"
              />
            </div>
            <h3
              class="jsx-723bb9ff0809bb37 font-medium text-lg mb-1"
            >
              Prof. Jibesh Patra
            </h3>
            <p
              class="jsx-723bb9ff0809bb37 text-sm text-gray-500 mb-2"
            >
              AI for SE, Software Security
            </p>
          </div>
          <div
            class="jsx-723bb9ff0809bb37 bg-white rounded-lg shadow-md p-6 flex flex-col items-center text-center transform transition-all duration-500 hover:-translate-y-2 hover:shadow-xl"
          >
            <div
              class="jsx-723bb9ff0809bb37 w-24 h-24 rounded-full bg-gray-200 mb-4 overflow-hidden"
            >
              <img
                alt="Prof. Animesh Mukherjee"
                class="jsx-723bb9ff0809bb37 w-full h-full object-cover"
                src="https://cse.iitkgp.ac.in/facimg/AMsmall.jpg?1743744045"
              />
            </div>
            <h3
              class="jsx-723bb9ff0809bb37 font-medium text-lg mb-1"
            >
              Prof. Animesh Mukherjee
            </h3>
            <p
              class="jsx-723bb9ff0809bb37 text-sm text-gray-500 mb-2"
            >
              Complex and Social Networks, Data and Web Mining
            </p>
          </div>
          <div
            class="jsx-723bb9ff0809bb37 bg-white rounded-lg shadow-md p-6 flex flex-col items-center text-center transform transition-all duration-500 hover:-translate-y-2 hover:shadow-xl"
          >
            <div
              class="jsx-723bb9ff0809bb37 w-24 h-24 rounded-full bg-gray-200 mb-4 overflow-hidden"
            >
              <img
                alt="Prof. Justin Solomon"
                class="jsx-723bb9ff0809bb37 w-full h-full object-cover"
                src="https://people.csail.mit.edu/jsolomon/assets/Justin_Solomon_3_small.jpg"
              />
            </div>
            <h3
              class="jsx-723bb9ff0809bb37 font-medium text-lg mb-1"
            >
              Prof. Justin Solomon
            </h3>
            <p
              class="jsx-723bb9ff0809bb37 text-sm text-gray-500 mb-2"
            >
              Machine Learning, Computer Vision, Graphs, Robotics
            </p>
          </div>
        </div>
        <div
          class="jsx-723bb9ff0809bb37 text-center mt-10"
        >
          <button
            class="jsx-723bb9ff0809bb37 px-6 py-3 bg-white border border-yellow-600 text-yellow-600 rounded-lg shadow-sm hover:bg-yellow-600 hover:text-white transition-all duration-300 transform hover:scale-105"
          >
            View All Professors
          </button>
        </div>
      </div>
    </section>
    <section
      class="jsx-723bb9ff0809bb37 py-16 bg-white"
      id="opport"
    >
      <div
        class="jsx-723bb9ff0809bb37 container mx-auto px-6"
      >
        <div
          class="jsx-723bb9ff0809bb37 flex flex-col md:flex-row items-center justify-between"
        >
          <div
            class="jsx-723bb9ff0809bb37 md:w-1/2 mb-10 md:mb-0"
          >
            <h2
              class="jsx-723bb9ff0809bb37 text-3xl font-bold mb-6 text-gray-800"
            >
              Latest Research Opportunities
            </h2>
            <p
              class="jsx-723bb9ff0809bb37 text-gray-600 mb-8 max-w-lg"
            >
              Stay updated with the newest research internships and projects available across various universities and research institutions.
            </p>
            <div
              class="jsx-723bb9ff0809bb37 space-y-4"
            >
              <div
                class="jsx-723bb9ff0809bb37 bg-gray-50 rounded-lg p-5 flex items-start border-l-4 border-yellow-600 transform transition-all duration-300 hover:-translate-x-1 hover:shadow-md"
              >
                <div
                  class="jsx-723bb9ff0809bb37 flex-1"
                >
                  <h3
                    class="jsx-723bb9ff0809bb37 font-medium text-lg"
                  >
                    Natural Language Processing
                  </h3>
                  <p
                    class="jsx-723bb9ff0809bb37 text-gray-500 text-sm mb-2"
                  >
                    NLP Lab, CSE Department
                  </p>
                  <div
                    class="jsx-723bb9ff0809bb37 flex items-center text-xs text-gray-500"
                  >
                    <span
                      class="jsx-723bb9ff0809bb37 mr-3"
                    >
                      Duration: 
                      3 months
                    </span>
                    <span
                      class="jsx-723bb9ff0809bb37"
                    >
                      Stipend: 
                      ₹25,000/month
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="jsx-723bb9ff0809bb37 bg-gray-50 rounded-lg p-5 flex items-start border-l-4 border-yellow-600 transform transition-all duration-300 hover:-translate-x-1 hover:shadow-md"
              >
                <div
                  class="jsx-723bb9ff0809bb37 flex-1"
                >
                  <h3
                    class="jsx-723bb9ff0809bb37 font-medium text-lg"
                  >
                    Computer Vision Research Internship
                  </h3>
                  <p
                    class="jsx-723bb9ff0809bb37 text-gray-500 text-sm mb-2"
                  >
                    AI Research Lab, CSE Department
                  </p>
                  <div
                    class="jsx-723bb9ff0809bb37 flex items-center text-xs text-gray-500"
                  >
                    <span
                      class="jsx-723bb9ff0809bb37 mr-3"
                    >
                      Duration: 
                      6 months
                    </span>
                    <span
                      class="jsx-723bb9ff0809bb37"
                    >
                      Stipend: 
                      ₹30,000/month
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="jsx-723bb9ff0809bb37 bg-gray-50 rounded-lg p-5 flex items-start border-l-4 border-yellow-600 transform transition-all duration-300 hover:-translate-x-1 hover:shadow-md"
              >
                <div
                  class="jsx-723bb9ff0809bb37 flex-1"
                >
                  <h3
                    class="jsx-723bb9ff0809bb37 font-medium text-lg"
                  >
                    Cybersecurity Research Internship
                  </h3>
                  <p
                    class="jsx-723bb9ff0809bb37 text-gray-500 text-sm mb-2"
                  >
                    Network Security Group, CSE Department
                  </p>
                  <div
                    class="jsx-723bb9ff0809bb37 flex items-center text-xs text-gray-500"
                  >
                    <span
                      class="jsx-723bb9ff0809bb37 mr-3"
                    >
                      Duration: 
                      6 months
                    </span>
                    <span
                      class="jsx-723bb9ff0809bb37"
                    >
                      Stipend: 
                      ₹35,000/month
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <button
              class="jsx-723bb9ff0809bb37 mt-8 px-6 py-3 bg-yellow-600 text-white rounded-lg shadow-md hover:bg-yellow-600 transition-all duration-300 transform hover:scale-105"
            >
              Browse All Opportunities
            </button>
          </div>
          <div
            class="jsx-723bb9ff0809bb37 md:w-2/5"
          >
            <div
              class="jsx-723bb9ff0809bb37 relative"
            >
              <div
                class="jsx-723bb9ff0809bb37 absolute -inset-4 bg-yellow-500 rounded-lg transform rotate-3 transition-transform duration-700 group-hover:rotate-6"
              />
              <img
                alt="Research"
                class="jsx-723bb9ff0809bb37 relative rounded-lg shadow-xl w-full h-auto"
                src="https://images.unsplash.com/photo-1542744173-8e7e53415bb0?q=80&w=2070&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
              />
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      class="jsx-723bb9ff0809bb37 py-20 bg-gradient-to-r from-yellow-600 to-purple-600 text-white"
    >
      <div
        class="jsx-723bb9ff0809bb37 container mx-auto px-6 text-center"
      >
        <h2
          class="jsx-723bb9ff0809bb37 text-4xl font-bold mb-6"
        >
          Ready to Begin Your Research Journey?
        </h2>
        <p
          class="jsx-723bb9ff0809bb37 text-xl mb-10 max-w-2xl mx-auto text-yellow-100"
        >
          Join thousands of students and professors already using our platform to connect and collaborate.
        </p>
        <div
          class="jsx-723bb9ff0809bb37 flex flex-col sm:flex-row justify-center space-y-4 sm:space-y-0 sm:space-x-6"
        >
          <button
            class="jsx-723bb9ff0809bb37 px-8 py-4 bg-white text-yellow-600 rounded-lg shadow-md hover:bg-gray-100 transition-all duration-300 transform hover:scale-105"
          >
            Sign In
          </button>
          <button
            class="jsx-723bb9ff0809bb37 px-8 py-4 bg-transparent border-2 border-white text-white rounded-lg shadow-md hover:bg-white hover:text-indigo-600 transition-all duration-300 transform hover:scale-105"
          >
            Create Account
          </button>
        </div>
      </div>
    </section>
    <footer
      class="jsx-723bb9ff0809bb37 bg-gray-900 text-gray-400 py-10"
    >
      <div
        class="jsx-723bb9ff0809bb37 container mx-auto px-6"
      >
        <div
          class="jsx-723bb9ff0809bb37 grid grid-cols-2 md:grid-cols-4 gap-8"
        >
          <div
            class="jsx-723bb9ff0809bb37"
          >
            <h3
              class="jsx-723bb9ff0809bb37 text-white font-medium mb-4"
            >
              For Students
            </h3>
            <ul
              class="jsx-723bb9ff0809bb37 space-y-2"
            >
              <li
                class="jsx-723bb9ff0809bb37"
              >
                <a
                  class="hover:text-white transition-colors duration-300"
                  href="#"
                >
                  Research Internships
                </a>
              </li>
              <li
                class="jsx-723bb9ff0809bb37"
              >
                <a
                  class="hover:text-white transition-colors duration-300"
                  href="#"
                >
                  Scholarships
                </a>
              </li>
            </ul>
          </div>
          <div
            class="jsx-723bb9ff0809bb37"
          >
            <h3
              class="jsx-723bb9ff0809bb37 text-white font-medium mb-4"
            >
              For Professors
            </h3>
            <ul
              class="jsx-723bb9ff0809bb37 space-y-2"
            >
              <li
                class="jsx-723bb9ff0809bb37"
              >
                <a
                  class="hover:text-white transition-colors duration-300"
                  href="#"
                >
                  Post Opportunities
                </a>
              </li>
              <li
                class="jsx-723bb9ff0809bb37"
              >
                <a
                  class="hover:text-white transition-colors duration-300"
                  href="#"
                >
                  Find Research Assistants
                </a>
              </li>
            </ul>
          </div>
          <div
            class="jsx-723bb9ff0809bb37"
          >
            <h3
              class="jsx-723bb9ff0809bb37 text-white font-medium mb-4"
            >
              Resources
            </h3>
            <ul
              class="jsx-723bb9ff0809bb37 space-y-2"
            >
              <li
                class="jsx-723bb9ff0809bb37"
              >
                <a
                  class="hover:text-white transition-colors duration-300"
                  href="https://www.connectedpapers.com/"
                  target="_blank"
                >
                  Academic Papers
                </a>
              </li>
              <li
                class="jsx-723bb9ff0809bb37"
              >
                <a
                  class="hover:text-white transition-colors duration-300"
                  href="https://fellowships.gov.in/Fellowships-Calculator"
                  target="_blank"
                >
                  Fellowships
                </a>
              </li>
            </ul>
          </div>
          <div
            class="jsx-723bb9ff0809bb37"
          >
            <h3
              class="jsx-723bb9ff0809bb37 text-white font-medium mb-4"
            >
              About
            </h3>
            <ul
              class="jsx-723bb9ff0809bb37 space-y-2"
            >
              <li
                class="jsx-723bb9ff0809bb37"
              >
                <a
                  class="hover:text-white transition-colors duration-300"
                  href="#"
                >
                  About Us
                </a>
              </li>
              <li
                class="jsx-723bb9ff0809bb37"
              >
                <a
                  class="hover:text-white transition-colors duration-300"
                  href="mailto:irisms.portal@gmail.com"
                >
                  Contact
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="jsx-723bb9ff0809bb37 border-t border-gray-800 mt-10 pt-6 flex flex-col md:flex-row justify-between items-center"
        >
          <p
            class="jsx-723bb9ff0809bb37"
          >
            © 2025 IRISMS
          </p>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
